src/Services/OrderService/OrderService.Application/UseCases/Orders/Queries/GetOrders/GetOrdersQueryHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using OrderService.Application.Common.Interfaces.Data;
+using OrderService.Application.UseCases.DTOs;
+
+namespace OrderService.Application.UseCases.Orders.Queries.GetOrders
+{
+    public class GetOrdersQueryHandler : IQueryHandler<GetOrdersQuery, GetOrdersResult>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public GetOrdersQueryHandler(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<GetOrdersResult> Handle(GetOrdersQuery query, CancellationToken cancellationToken)
+        {
+            var pageIndex = query.PaginationRequest.PageIndex;
+            var pageSize = query.PaginationRequest.PageSize;
+
+            var allOrders = await _unitOfWork.OrderRepository.GetAllOrdersAsync(cancellationToken);
```

Comment:
## 🚨 Critical Error

this is wrong implementation of paginated db query, you have to pass skip amount and take amount to repository method, which should pass to db, which will return exactly the right amount of entities

your method returns all entities from db to RAM which will break down the server